packages/service-library/src/servicelib/aiohttp/tracing.py

```diff
@@ -1,76 +1,68 @@
-""" Adds aiohttp middleware for tracing using zipkin server instrumentation.
+""" Adds aiohttp middleware for tracing using opentelemetry instrumentation.
 
 """
 import logging
-from typing import Iterable
 
-import aiozipkin as az
 from aiohttp import web
-from aiohttp.web import AbstractRoute
-from aiozipkin.aiohttp_helpers import (
-    APP_AIOZIPKIN_KEY,
-    REQUEST_AIOZIPKIN_KEY,
-    middleware_maker,
+from opentelemetry import trace
+from opentelemetry.exporter.otlp.proto.http.trace_exporter import (
+    OTLPSpanExporter as OTLPSpanExporterHTTP,
 )
-from yarl import URL
+from opentelemetry.instrumentation.aiohttp_client import AioHttpClientInstrumentor
+from opentelemetry.instrumentation.aiohttp_server import AioHttpServerInstrumentor
+from opentelemetry.instrumentation.aiopg import AiopgInstrumentor
+from opentelemetry.instrumentation.requests import RequestsInstrumentor
+from opentelemetry.sdk.resources import Resource
+from opentelemetry.sdk.trace import TracerProvider
+from opentelemetry.sdk.trace.export import BatchSpanProcessor
+from settings_library.tracing import TracingSettings
 
 log = logging.getLogger(__name__)
 
 
 def setup_tracing(
-    app: web.Application,
-    *,
+    app: web.Application,  # pylint: disable=unused-argument
+    tracing_settings: TracingSettings,
     service_name: str,
-    host: str,
-    port: int,
-    jaeger_base_url: URL | str,
-    skip_routes: Iterable[AbstractRoute] | None = None,
-) -> bool:
+    instrument_aiopg: bool = False,  # noqa: FBT001, FBT002
+) -> None:
     """
-    Sets up this service for a distributed tracing system
-    using zipkin (https://zipkin.io/) and Jaeger (https://www.jaegertracing.io/)
+    Sets up this service for a distributed tracing system (opentelemetry)
     """
-    zipkin_address = URL(f"{jaeger_base_url}") / "api/v2/spans"
-
-    log.debug(
-        "Setting up tracing for %s at %s:%d -> %s",
-        service_name,
-        host,
-        port,
-        zipkin_address,
+    opentelemetry_collector_endpoint = (
+        tracing_settings.TRACING_OPENTELEMETRY_COLLECTOR_ENDPOINT
     )
-
-    endpoint = az.create_endpoint(service_name, ipv4=host, port=port)
-
-    # TODO: move away from aiozipkin to OpenTelemetrySDK
-    # https://opentelemetry-python-contrib.readthedocs.io/en/latest/instrumentation/asgi/asgi.html
-    # see issue [#2715](https://github.com/ITISFoundation/osparc-simcore/issues/2715)
-    # creates / closes tracer
-    async def _tracer_cleanup_context(app: web.Application):
-
-        app[APP_AIOZIPKIN_KEY] = await az.create(
-            f"{zipkin_address}", endpoint, sample_rate=1.0
+    opentelemetry_collector_port = tracing_settings.TRACING_OPENTELEMETRY_COLLECTOR_PORT
+    if not opentelemetry_collector_endpoint and not opentelemetry_collector_port:
```

Comment:
NOTE: Generally "skipping" setup we do it differently. It is normally done in the app level settings by nullifying the entire env var e.g. in the webserver we would just do `WEBSERVER_TRACING=null` and the `setup` is automatically skipped.

But nevermind... this is ok like this for now. I might refactor a bit in the near future